tools/isobuild/import-scanner.js

```diff
@@ -825,14 +818,11 @@ export default class ImportScanner {
     // Set file.imported to a truthy value (either "dynamic" or true).
     file.imported = forDynamicImport ? "dynamic" : true;
 
-    if (file.error) {
+    if (file.reportPendingErrors &&
+        file.reportPendingErrors() > 0) {
```

Comment:
This tends to be the first place the `ImportScanner` forces full compilation of `file`.